tests/unit/intersection/test_policy_shard.py

```diff
@@ -0,0 +1,242 @@
+                {Condition(key="aws:PrincipalOrgId", operator="StringNotEquals", values=["o-123456"])}
+            ),
+            not_conditions=frozenset(),
+        ),
+        "second": PolicyShard(
+            effect="Allow",
+            effective_action=EffectiveAction(inclusion=Action("s3:*"), exclusions=frozenset()),
+            effective_resource=EffectiveResource(inclusion=Resource("*"), exclusions=frozenset()),
+            effective_principal=EffectivePrincipal(inclusion=Principal(type="AWS", value="*"), exclusions=frozenset()),
+            conditions=frozenset(
+                {
+                    Condition(key="aws:PrincipalOrgId", operator="StringNotEquals", values=["o-123456"]),
+                    Condition(key="s3:x-amz-server-side-encryption", operator="StringEquals", values=["AES256"]),
+                }
+            ),
+            not_conditions=frozenset(),
+        ),
+        "result": PolicyShard(
+            effect="Allow",
+            effective_action=EffectiveAction(inclusion=Action("s3:*"), exclusions=frozenset({Action("s3:PutObject")})),
+            effective_resource=EffectiveResource(inclusion=Resource("*"), exclusions=frozenset()),
+            effective_principal=EffectivePrincipal(inclusion=Principal(type="AWS", value="*"), exclusions=frozenset()),
+            conditions=frozenset(
+                {Condition(key="aws:PrincipalOrgId", operator="StringNotEquals", values=["o-123456"])}
+            ),
+            not_conditions=frozenset(),
+        ),
+    },
+    "test_matching_subset_conditions_smaller_first": {
+        "first": PolicyShard(
+            effect="Allow",
+            effective_action=EffectiveAction(inclusion=Action("s3:*"), exclusions=frozenset()),
+            effective_resource=EffectiveResource(inclusion=Resource("*"), exclusions=frozenset()),
+            effective_principal=EffectivePrincipal(inclusion=Principal(type="AWS", value="*"), exclusions=frozenset()),
+            conditions=frozenset(
+                {
+                    Condition(key="aws:PrincipalOrgId", operator="StringNotEquals", values=["o-123456"]),
+                    Condition(key="s3:x-amz-server-side-encryption", operator="StringEquals", values=["AES256"]),
+                }
+            ),
+            not_conditions=frozenset(),
+        ),
+        "second": PolicyShard(
+            effect="Allow",
+            effective_action=EffectiveAction(inclusion=Action("s3:*"), exclusions=frozenset({Action("s3:PutObject")})),
+            effective_resource=EffectiveResource(inclusion=Resource("*"), exclusions=frozenset()),
+            effective_principal=EffectivePrincipal(inclusion=Principal(type="AWS", value="*"), exclusions=frozenset()),
+            conditions=frozenset(
+                {Condition(key="aws:PrincipalOrgId", operator="StringNotEquals", values=["o-123456"])}
+            ),
+            not_conditions=frozenset(),
+        ),
+        "result": PolicyShard(
```

Comment:
Does this result make sense?
In calculating the intersection it removes a condition from shard A.
This means it's INCLUDING scenarios that were previously excluded, i.e. the intersection contains things which weren't in the original set.

Reply:
No, it doesn't, updated to reflect this.